modal/_utils/task_command_router_client.py

```diff
@@ -205,29 +211,40 @@ async def send_request(event: grpclib.events.SendRequest) -> None:
         self._stub = TaskCommandRouterStub(self._channel)
 
     def __del__(self) -> None:
-        """Clean up the client when it's garbage collected."""
-        if self._closed:
+        """Best-effort cleanup if the caller forgot to close().
+
+        This object is typically used through synchronicity wrappers, which means this finalizer
+        may run on a different thread than the event loop that owns the channel. Closing the
+        channel is therefore scheduled onto the owning loop using call_soon_threadsafe.
+
+        Use getattr in the event that attributes are not yet initialized or the
+        object is in a half-torn-down state.
+        """
+        if getattr(self, "_closed", False):
```

Comment:
This feels a bit too defensive? If we initialize these in the constructor then they should always be set right

Reply:
`__del__` is [called](https://stackoverflow.com/questions/6409644/is-del-called-on-an-object-that-doesnt-complete-init) even when `__init__` throws, so you can end up with a partially initialized object